src/features/wizard/WizardSlice.ts

```diff
@@ -147,26 +169,69 @@ const getIndexAndParentPathOfUISchemaElement: (uiSchemaPath: string) => {
   }
 }
 
-const prepareRemoveLayoutFromUISchema: (
-  uiSchema: UISchemaElement,
-  uiSchemaPath: string
-) => { pointer: string; elements: UISchemaElement[] } = (uiSchema, uiSchemaPath) => {
-  const { index, parentPath } = getIndexAndParentPathOfUISchemaElement(uiSchemaPath)
-  if (index === undefined || !parentPath) {
-    throw new Error('Invalid path, path should lead to an element part of an array, thus be at least 2 segments long')
-  }
-  const pointer = pathSegmentsToJSONPointer(pathToPathSegments(parentPath))
-  const oldUISchema = jsonpointer.get(uiSchema, pointer)
-  if (!Array.isArray(oldUISchema)) {
-    throw new Error('UISchema Elements bucket is not an array')
+const getUiSchemaOfDraggable: (
+  draggableComponent: DraggableComponent | DraggableUISchemaElement,
+  newKey: string
+) => UISchemaElement | undefined = (draggableComponent, newKey) => {
+  const { name, uiSchema } = draggableComponent
+
+  if (draggableComponent.uiSchema) {
+    return updateScopeOfUISchemaElement(
+      `#/properties/${name}`, //this does only apply to newly created elements
+      pathSegmentsToScope([newKey]),
+      uiSchema
+    )
+  } else {
+    return {
+      ...(uiSchema || {}),
+      type: 'Control',
+      scope: pathSegmentsToScope([newKey]),
+    }
   }
+}
+
+const getUiSchemaWithFlatScope: (
+  draggableComponent: DraggableComponent | DraggableUISchemaElement,
+  deepestGroupPath: string[],
+  newKey: string
+) => UISchemaElement | undefined = (draggableComponent, deepestGroupPath, newKey) => {
+  const { name, uiSchema } = draggableComponent
 
   return {
-    pointer,
-    elements: [...oldUISchema.slice(0, index), ...oldUISchema.slice(index + 1)],
+    type: 'Control',
+    ...(uiSchema || {}),
+    scope: pathSegmentsToScope([...deepestGroupPath, newKey]),
   }
 }
 
+const getDeepestGroupPath = (structurePath: string, uiSchema: any): string[] => {
```

Comment:
hier musste ich eine ganze Weie lesen, bis ich verstanden habe was hier passiert, das ist also auch um dann mal zu berechnen in welche Ebene das Element im json schema eingefügt werden soll?

Reply:
genau, mit der dokumentation wollte ich warten bis ich eine funktionierende implementation habe. Das Thema ist ja nicht trivial und braucht denke ich später mal einen eigenen pull request. Zurzeit ist diese Funktion nicht in Benutzung.